wordpress-mobile/WordPress-Android
# Remove Site Name experiment code and set feature flag to disabled

Fixes #16748

### Description

This removes the experiment code (A/B test) for the Site Name feature, and disables the feature flag. It also removes the associated tracking events, which are no longer useful.

### To test:

1. Start the site creation flow
2. Navigate through the experience to the last step (choosing a domain)
3. Ensure that the site name screen is not displayed and that the choose a domain screen _is_ displayed

Note of caution:

In my testing, I observed that even though the default value for the feature flag has been set to `false`, it was still set to `true` on my testing device. I'm not sure if this is because I had previously overridden the default value of `false` by manually setting it to `true` in an earlier phase of the project, or if the default state of `true` has remained somehow set.

I will try to investigate the issue described above, since it wouldn't be desirable to have the feature enabled for users who happened to install the app during the enabled phase of the experiment.

## Regression Notes
1. Potential unintended areas of impact
N/A

4. What I did to test those areas of impact (or what existing automated tests I relied on)


5. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
